### PR TITLE
Separate withTheme from Button component

### PR DESCRIPTION
**Pull request by wEprous**
Separated withTheme from Button component.
Deleted ThemeProvider and unnecessary options from stories.
Updated test.

**No more changes**

### CODE REVIEW

**Comment by tomashapl** [comment]
@darkwindcz Should `<Button />` render <a> like in `<ButtonLink />` ?

**Comment by darkwindcz** [comment]
@tomashapl yes,  both `<Button />` and `<ButtonLink />` should be able to render `a` or `button`.

In my opinion, when developer use prop `href`, it should render `button`. I am not sure if it's needed to be able to tell component what type it should render manually.

**Comment by tomashapl** [comment]
@darkwindcz Thanks.

@wEprous Can you handle this ? It should be the same like in ButtonLink component i think.

**Comment by wEprous** [comment]
@tomashapl Yeah, sure :)

**Comment by tomashapl** [comment]
@wEprous 

You forgot delete old code from `Button/index.js` right ? please check your code before pushing it to review. 🙏

File: src/Button/Button.js [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+  disabled: boolean,
+  block: boolean,
+  type: $Values<typeof TYPE_OPTIONS>,
+  size: $Values<typeof SIZE_OPTIONS>,
+  width: number,
+  icon?: React.Node,
+  theme: typeof defaultTokens,
+|};
+
+const IconContainer = styled(({ theme, tokens, bordered, sizeIcon, type, onlyIcon, ...props }) => (
```
Comment (tomashapl) [inline src/Button/Button.js]: I think there isn't available any other props

why there you need it ?

```
<IconContainer
          size={size}
          onlyIcon={onlyIcon}
          bordered={bordered}
          sizeIcon={sizeIcon}
          tokens={tokens}
          type={type}
          theme={theme}
        >
```


just render `<div />` without props

Comment (wEprous) [reply src/Button/Button.js]: We need to pass at least `children` and `className` to this div.

Comment (tomashapl) [reply src/Button/Button.js]: OK, so i think this is more readable, and you shouldn't destruct unused props

image

**Pull request merged**